scroll-search.lic

```diff
@@ -0,0 +1,520 @@
+
+  def get_ordinal
+    return @ordinal
+  end
+  
+  def get_index
+	return @index
+  end
+  
+  # represents the local index of the
+  # scroll within the container
+  def get_container_index
+	return @container_index
+  end
+
+  def get_type
+    return @type
+  end
+
+  def set_container(container)
+    @container = container
+  end
+  
+  def set_container_index(index)
+	@container_index = index
+  end
+  
+  def set_ordinal(ordinal)
+	@ordinal = ordinal
+  end
+end
+
+class ScrollSearch
+  include DRC
+  include DRCI
+
+  def initialize
+
+    # Variables
+    @scroll_count = 0
+	@global_scroll_index = if UserVars.scroll_counter.nil? then 0 else UserVars.scroll_counter end
+    @owned_scrolls = if UserVars.owned_scrolls.empty? then {} else UserVars.owned_scrolls end
+    
+	# RESET ALL
+	# UserVars.owned_scrolls = { }
+	# @owned_scrolls = { }
+	# UserVars.scroll_counter = 0
+
+    $settings = get_settings()
+    $debug_mode = $settings.debug_mode
+
+    echo("Settings debug mode is: #{$debug_mode}") if $debug_mode
+    if $settings.nil? || $settings.empty?
+      echo("Settings was nil!") if $debug_mode
+    end
+
+    arg_definitions = 
+    [
+      [
+		{ name: 'search', regex: /^sea.*/i, optional: true, description: 'Searches given container and labels any unlabeled scrolls. ' },
+		{ name: 'list', regex: /^lis.*/i, optional: true, description: 'Lists all of the scrolls that have been categorized after a SEARCH. ' },
+        { name: 'container', regex: /\w+/i, optional: true, description: 'The container to search. ' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+
+	unless args.search || args.list
+		display_args(arg_definitions)
+		return
+	end
+
+    pause
+
+    if args.search
+      if args.container.nil?
+        echo "*** SPECIFY THE CONTAINER TO SEARCH! ***"
+        return
+      else
+        search_container(args.container)
+      end
+	elsif args.list
+      if @owned_scrolls.length <= 0
+        echo("*** NO SCROLLS HAVE BEEN ADDED! SEARCH YOUR CONTAINERS FIRST!***")
+        return
+      end
+
+      echo("Scroll scout is: #{@owned_scrolls.length}") if $debug_mode
+      display_scrolls()
+    else
+      echo("***INVALID ACTION***")
+    end
+  end
+
+  def search_container(container)
+    # return if invalid container
+
+	unless exists?(container)
+      echo("*** INVALID CONTAINER. ***")
+      return
+    end
+
+    # Get long name of container
+    $container_full = full_name(container)
+    $container = get_noun($container_full)
+
+    # iterate over all scroll types
+    scroll_types = $settings.scroll_nouns
+
+    if scroll_types.length <= 0 || scroll_types.nil?
+      echo("*** NO SCROLL TYPTES SET! SETUP YOUR YAML TO INCLUDE VALID \"scroll_types\". ***")
+      return
+    end
+
+    echo("Scroll type count is: #{scroll_types.length}.") if $debug_mode
+
+	# Clear current scroll listing from owned scrolls if
+	# user is searching an existing container
+	if @owned_scrolls.key? ($container_full)

```

Comment:
You can use truthy falsey values to simplify this to `if @owned_scrolls[$container_full]`. Also it would be nice to move away from global vars.